core/arch/arm/kernel/spmc_sp_handler.c

```diff
@@ -219,10 +219,54 @@ static int spmc_sp_add_sp_region(struct sp_mem *smem,
 	return res;
 }
 
-static int spmc_sp_add_nw_region(struct sp_mem *smem __unused,
-				 struct ffa_mem_region *mem_reg __unused)
+static int spmc_sp_add_nw_region(struct sp_mem *smem,
+				 struct ffa_mem_region *mem_reg)
 {
+	uint64_t page_count = READ_ONCE(mem_reg->total_page_count);
+	struct mobj *mobj = NULL;
+	struct sp_mem_map_region *region = NULL;
+	struct mobj *m = sp_mem_new_mobj(page_count);
+	unsigned int i = 0;
+	unsigned int idx = 0;
+	int res = FFA_OK;
+	uint64_t address_count = READ_ONCE(mem_reg->address_range_count);
+
+	if (!m)
+		return FFA_DENIED;
+
+	for (i = 0; i < address_count; i++) {
+		struct ffa_address_range *addr_range = NULL;
+
+		addr_range =  &mem_reg->address_range_array[i];
+		if (sp_mem_add_pages(m, &idx,
+				     READ_ONCE(addr_range->address),
+				     READ_ONCE(addr_range->page_count))) {
+			res = FFA_DENIED;
+			goto clean_up;
+		}
+	}
+
+	region = malloc(sizeof(*region));
+	if (!region) {
+		res = FFA_NO_MEMORY;
+		goto clean_up;
+	}
+
+	region->mobj = m;
+	region->page_offset = 0;
```

Comment:
(For commit: "core: FFA_SHARE: Process Normal World share")
prefer `calloc()` and remove field specific 0 value initialization.

Reply:
Done